### PR TITLE
Update the Android SDK version and the test apps 

### PR DESCRIPTION
**Pull request by russelarms**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

- Update the minimal Android SDK version number.
- Update test apps

## Related PRs or issues

[AB#83684](https://msmobilecenter.visualstudio.com/Mobile-Center/_boards/board/t/Ivan-Team/Backlog%20Items/?workitem=83684)

## Misc

The target SDK version is 29 as it is currently used for newly created RN Android apps.

### CODE REVIEW

**Comment by russelarms** [comment]
Jest tests will be fixed in another PR.

File: AppCenterReactNativeShared/android/build.gradle [hunk 1]
```diff
@@ -28,12 +28,12 @@ allprojects {
...
-        compileSdkVersion 27
-        buildToolsVersion '27.0.3'
+        compileSdkVersion 29
+        buildToolsVersion '29.0.2'
 
         defaultConfig {
-            minSdkVersion 16
-            targetSdkVersion 26
+            minSdkVersion 21
+            targetSdkVersion 29
```
Comment (MatkovIvan) [inline AppCenterReactNativeShared/android/build.gradle:36]: 30?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 3.1.3
 
+#### Android
+
+* **[Breaking change]** Bump the minimum Android SDK version to API level 21.
```
Comment (MatkovIvan) [inline CHANGELOG.md:7]: Copy wording from Android

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: AppCenterReactNativeShared/android/build.gradle (hunk 1 above)
Comment (russelarms) [reply AppCenterReactNativeShared/android/build.gradle:36]: The target SDK version is 29 as it is currently used for newly created RN Android apps.

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**